modules/fedimint-ln-client-ng/src/lib.rs

```diff
@@ -0,0 +1,449 @@
+use std::collections::BTreeMap;
+use std::sync::Arc;
+use std::time::Duration;
+
+use async_stream::stream;
+use fedimint_client::module::init::{ClientModuleInit, ClientModuleInitArgs};
+use fedimint_client::module::recovery::NoModuleBackup;
+use fedimint_client::module::{ClientContext, ClientModule, IClientModule};
+use fedimint_client::oplog::UpdateStreamOrOutcome;
+use fedimint_client::sm::util::MapStateTransitions;
+use fedimint_client::sm::{DynState, ModuleNotifier, State, StateTransition};
+use fedimint_client::transaction::{ClientOutput, TransactionBuilder};
+use fedimint_client::{sm_enum_variant_translation, DynGlobalClientContext};
+use fedimint_core::api::DynModuleApi;
+use fedimint_core::core::{IntoDynInstance, ModuleInstanceId, OperationId};
+use fedimint_core::db::DatabaseTransaction;
+use fedimint_core::encoding::{Decodable, Encodable};
+use fedimint_core::module::{
+    ApiVersion, CommonModuleInit, ModuleCommon, ModuleInit, MultiApiVersion, TransactionItemAmount,
+};
+use fedimint_core::task::sleep;
+use fedimint_core::{apply, async_trait_maybe_send, OutPoint};
+use fedimint_ln_common_ng::api::LnFederationApi;
+use fedimint_ln_common_ng::contracts::{
+    ContractKey, DecryptionContract, FundedDecryptionContract, LightningContract,
+};
+use fedimint_ln_common_ng::{
+    LightningClientContext, LightningCommonInit, LightningModuleTypes, LightningOutput,
+};
+use futures::StreamExt;
+use secp256k1::KeyPair;
+use serde::{Deserialize, Serialize};
+use tpe::AggregateDecryptionKey;
+use tracing::error;
+
+use crate::internal_send::{Funding, InternalSendSMState, InternalSendStateMachine};
+use crate::lightning_send::{LightningSendSMState, LightningSendStateMachine};
+use crate::receive::{Pending, ReceiveSMState, ReceiveStateMachine};
+
+mod internal_send;
+mod lightning_send;
+mod receive;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct LightningOperationMeta;
+
+const RETRY_DELAY: Duration = Duration::from_secs(1);
+
+/// The high-level state of an payment operation over lightning,
```

Comment:
Just adding this comment so I don't forget (I know its a nit), but we'll need to add more metadata to these high level states